pkg/operation/botanist/botanist.go

```diff
@@ -199,10 +199,13 @@ func (b *Botanist) RequiredExtensionsExist() error {
 func (b *Botanist) computeRequiredExtensions() map[string]string {
 	requiredExtensions := map[string]string{
 		extensionsv1alpha1.OperatingSystemConfigResource: string(b.Shoot.GetMachineImageName()),
-		dnsv1alpha1.DNSProviderKind:                      b.Garden.InternalDomain.Provider,
 	}
 
-	if b.Shoot.ExternalDomain != nil {
+	if b.Garden.InternalDomain.Provider != gardenv1beta1.DNSUnmanaged {
+		requiredExtensions[dnsv1alpha1.DNSProviderKind] = b.Garden.InternalDomain.Provider
```

Comment:
BTW, shouldn't the key-value-mapping the other way around, i.e. `map[type]kind` since the cardinality of `kind:type` is `1:n`?

I know this is not addressed by this PR but I just noticed it incidentally. /cc @rfranzke

Reply:
Good catch, yes, let's address this separately